### ISSUE TITLE
Failed to reproduce DM_BOXED_PRIMITIVE_FOR_COMPARE

### ISSUE THREAD

**Issue by Xiaoven** [spotbugs/spotbugs#1835]
# Description
I wanted to reproduce DM_BOXED_PRIMITIVE_FOR_COMPARE warnings in [Feature333.java](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugsTestCases/src/java/sfBugsNew/Feature333.java) with SpotBugs Maven plugin, but failed.  

An example is:
```java
@ExpectWarning("DM_BOXED_PRIMITIVE_FOR_COMPARE")
    public int compareTo(long a, long b) {
        return ((Long)a).compareTo(b);
    }
```

## Configuration
```xml
        <plugin>
            <groupId>com.github.spotbugs</groupId>
            <artifactId>spotbugs-maven-plugin</artifactId>
            <version>4.4.2.2</version>
            <dependencies>
                <!-- overwrite dependency on spotbugs if you want to specify the version of spotbugs -->
                <dependency>
                    <groupId>com.github.spotbugs</groupId>
                    <artifactId>spotbugs</artifactId>
                    <version>4.5.0</version>
                </dependency>
            </dependencies>
            <configuration>
                <effort>Default</effort>
                <threshold>Low</threshold>
                <debug>true</debug>
            </configuration>
        </plugin>
```
# What I have done
I read the source code to detect this pattern in [DumbMethods.java](https://github.com/spotbugs/spotbugs/blob/21248ad66903687ab443b74b73e8e722c0cf6fb1/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java#L752), and find this pattern is under a branch **which conditions seemed unrelated to this pattern**.  The conditions are :
```java
if ("compareTo".equals(called.getName())
                        && "valueOf".equals(previousMethodCall.getName())
                        && called.getClassDescriptor().equals(previousMethodCall.getClassDescriptor()) && !previousMethodCall.getSignature()
                                .startsWith("(Ljava/lang/String;")) {
```

I don't think this pattern requires the presence of `valueOf` method.
